### ISSUE TITLE
Upgrading a volume-persisted Ignition Container 7.9.13 to 7.9.14

### ISSUE THREAD

**Issue by mjohncox** [thirdgen88/ignition-docker#19]
Made a gateway in version 7.9.13 connected to persistent volume at launch.
Updated container image to 7.9.14 and redeployed.

Error received:
Detected Ignition Volume from prior version (7.9.13), running Upgrader                                                                                                                         │
│ Error: Could not find or load main class com.inductiveautomation.ignition.common.upgrader.Upgrader

**Comment by thirdgen88** [comment]
I see, interesting..  I will look into this and let you know what I find!  Thanks for the heads up!

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by mjohncox** [comment]
Thanks.  I like the simplicity of this approach.  For now, just deleting the pvc, redeploy 7.9.14 gateway, and restore backup from 7.9.13 seems to have worked ok.

**Comment by ptrdlbrg** [comment]
I ran into the same problem upgrading from 7.9.12 to 7.9.14. The problem is that the `docker-entrypoint.sh` script is attempting to run the `Upgrader` class from a jar which isn't present. In particular the script references the`"lib/core/common/common-${image_version}.jar"`. If you check the `/usr/local/share/ignition/lib/core/common` directory you'll see that there is a `common-7.9.14.jar.pack.gz` and `common.jar` but no `common-7.9.14.jar`.

The `common.jar` seems to be the only .jar file with the version stripped from its name.

**Comment by thirdgen88** [comment]
Just looked into this, and @ptrdlbrg is correct.  I'm adding an extra step to the Dockerfile to check for the `.pack.gz` of this `common-${IGNITION_VERSION}.jar` file and unpacking it if necessary.

**Comment by ptrdlbrg** [comment]
@thirdgen88 
I was thinking that the solution would be to change the `docker-entrypoint.sh` file to check if the `common-${IGNITION_VERSION}.jar` exists and if it doesn't then run the Upgrader from the `common.jar` instead. That avoids the overhead of an additional jar and an extra docker image layer.

Of course, I may be oblivious to the reasons why you're unpacking the jar. So maybe my idea is missing something important.

**Comment by thirdgen88** [comment]
>@thirdgen88  ...

@ptrdlbrg I can certainly try that—it would be a little leaner.  Alternatively I suppose I could also just do that `unpack200` on-demand if necessary within the entrypoint..  I know it is named the same, but do we know for sure that the root `common.jar` is the same?  I haven’t tried to interrogate the classes available in there yet.  They are slightly different file sizes (the unpacked common-${IGNITION_VERSION}.jar and the common.jar).

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by ptrdlbrg** [comment]
I did notice that the jars are slightly different sizes. I'm supposing that's an artifact of the packing/unpacking process. When I did a diff of the zip entries in both jar files, they came out to be the same except the pack200 version, `common-7.9.14.jar`, was missing directory entries. 

Those directory entries aren't important for a jar file so it makes sense to remove them to save space. If you unzip the jar files then the `unzip` program will automatically create the missing directories. I did a comparison of the contents of the two unzipped jar files and the two sets of file names were identical.

**Comment by thirdgen88** [comment]
I decided to review the inspiration for the upgrade integration we're discussing now, and in referencing the README (for a few past versions) of the zip installer, I see that the guidance there directs running the upgrader from the `common.jar`; however, in previous releases, there was no `common.jar` in that location, only the `common-{IGNITION_VERSION}.jar.pack.gz` and the `common-{IGNITION_VERSION}.jar`.  Therefore, I just targeted that one and life was good.  It seems that now the `common.jar` that is actually directly referenced in the README is there as it should be.  So at this point, I think I'm just going to simplify (which also aligns it with the 8.0 branch) and just call the upgrader against the `common.jar`, no need to do any dancing anymore at all!
